app/src/main/cpp/skyline/applet/swkbd/software_keyboard_config.h

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2022 Skyline Team and Contributors (https://github.com/skyline-emu/)
+// Copyright © 2019-2022 Ryujinx Team and Contributors
+
+#pragma once
+
+#include <services/account/IAccountServiceForApplication.h>
+
+namespace skyline::applet::swkbd {
+    /**
+     * @brief Specifies the characters the keyboard should allow you to input
+     * @url https://switchbrew.org/wiki/Software_Keyboard#KeyboardMode
+     */
+    enum class KeyboardMode : u32 {
+        Full = 0x0,
+        Numeric = 0x1,
+        ASCII = 0x2,
+        FullLatin = 0x3,
+        Alphabet = 0x4,
+        SimplifiedChinese = 0x5,
+        TraditionalChinese = 0x6,
+        Korean = 0x7,
+        LanguageSet2 = 0x8,
+        LanguageSet2Latin = 0x9,
+    };
+
+    /**
+     * @brief Specifies the characters that you shouldn't be allowed to input
+     * @url https://switchbrew.org/wiki/Software_Keyboard#InvalidCharFlags
+     */
+    union InvalidCharFlags {
+        u32 raw;
+        struct {
+            u32 _pad_ : 1;
+            u32 space : 1;
+            u32 atMark : 1;
+            u32 percent : 1;
+            u32 slash : 1;
+            u32 backslash : 1;
+            u32 numeric : 1;
+            u32 outsideOfDownloadCode : 1;
+            u32 outsideOfMiiNickName : 1;
+        } flags;
+    };
+
+    /**
+     * @brief Specifies where the cursor should initially be on the initial string
+     * @url https://switchbrew.org/wiki/Software_Keyboard#InitialCursorPos
+     */
+    enum class InitialCursorPos : u32 {
+        First = 0x0,
+        Last = 0x1,
+    };
+
+    /**
+     * @url https://switchbrew.org/wiki/Software_Keyboard#PasswordMode
+     */
+    enum class PasswordMode : u32 {
+        Show = 0x0,
+        Hide = 0x1, //!< Hides any inputted text to prevent a password from being leaked
+    };
+
+    /**
+     * @url https://switchbrew.org/wiki/Software_Keyboard#InputFormMode
+     * @note Only applies when 1<= textMaxLength <= 32, otherwise Multiline is used
```

Comment:
Space after `1` but good job with comments in this file, they're to the point and fairly useful.